src/NuGet.Core/NuGet.Commands/Utility/SolutionPackagesContentHashUtility.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Commands.Utility
+{
+    public class SolutionPackagesContentHashUtility : ISolutionPackagesContentHashUtility
+    {
+        private readonly SourceRepository _packagesFolderSourceRepository;
+        private readonly IReadOnlyList<SourceRepository> _globalPackageFolderRepositories;
+        private readonly ILogger _logger;
+
+        public SolutionPackagesContentHashUtility(SourceRepository packagesFolderSourceRepository, IReadOnlyList<SourceRepository> globalPackageFolderRepositories, ILogger logger)
+        {
+            if (!packagesFolderSourceRepository.PackageSource.IsLocal)
+            {
+                throw new ArgumentException($"Internal error. Package source {packagesFolderSourceRepository.PackageSource.Name} is not local");
+            }
+
+            foreach (var repository in globalPackageFolderRepositories)
+            {
+                if (!repository.PackageSource.IsLocal)
+                {
+                    throw new ArgumentException($"Internal error. Package source {repository.PackageSource.Name} is not local");
+                }
+            }
+
+            _packagesFolderSourceRepository = packagesFolderSourceRepository;
+            _globalPackageFolderRepositories = globalPackageFolderRepositories;
```

Comment:
global package folder repositorie**s**.
Multiple? 
How?

Reply:
fallback folders were in the list when I was debugging. I'm not sure if it adds all local folders, or if the fallback folder was somehow special, but it was there.